snowplow_tracker/emitters.py

```diff
@@ -153,6 +156,11 @@ def __init__(
         self.custom_retry_codes = custom_retry_codes
         logger.info("Emitter initialized with endpoint " + self.endpoint)
 
+        if session is None:
+            session = requests
+
+        self.request_method = session
```

Comment:
Nit: instead of reassigning the session object, this would look a bit more readable to me:

```suggestion
        self.request_method = requests if session is None else session
```